File: src/kyu_6/reverse_or_rotate/cpp/solution.cpp
#include <string>
#include <algorithm>

class RevRot
{
public:
    static std::string revRot(const std::string &strng, unsigned int sz){
      if(sz <= 0 || strng.empty() || sz > strng.length())
        return "";
      std::string x = strng.substr(0,sz);
      bool p = true;
      for(const char& c : x)
        if(c % 2 == 1) // '0' == 48, so 'digit' is even iff digit is even
          p = !p;
      if(p){
        std::reverse(x.begin(),x.end());
        return x + revRot(strng.substr(sz),sz);
      }
      x += x[0];
      return x.substr(1) + revRot(strng.substr(sz),sz);
    }
};

File: src/kyu_6/reverse_or_rotate/csharp/solution.cs
using System;
using System.Linq;

public class Revrot
{
    public static string RevRot(string strng, int sz)
    {
        if (String.IsNullOrEmpty(strng) || sz <= 0 || sz > strng.Length)
            return String.Empty;

        return
            new String(
                Enumerable.Range(0, strng.Length/sz)
                    .Select(i => strng.Substring(i*sz, sz))
                    .Select(
                        chunk =>
                            chunk.Sum(digit => (int) Math.Pow(int.Parse(digit.ToString()), 3))%2 == 0
                                ? chunk.Reverse()
                                : chunk.Skip(1).Concat(chunk.Take(1)))
                    .SelectMany(x => x)
                    .ToArray());
    }
}